core/libp2p/multi/multistream.hpp

```diff
@@ -6,11 +6,66 @@
 #ifndef KAGOME_MULTISTREAM_HPP
 #define KAGOME_MULTISTREAM_HPP
 
+#include <filesystem>
+#include <gsl/span>
+
+#include "common/buffer.hpp"
+#include "common/result.hpp"
+
 namespace libp2p::multi {
   /**
    * Format of stream identifier used in Libp2p
+   * @see https://github.com/multiformats/multistream
    */
-  class Multistream {};
+  class Multistream {
+   public:
+    /**
+     * A protocol used in the stream is represented as a UNIX URI instead of just its name,
+     * since it's much more descriptive
+     */
+    using Path = std::filesystem::path;
```

Comment:
As I see, `std::filesystem::path` is only about paths on the machine, but not about protocols, such as Json or http. If I am right, is there any sense in using it instead of just name?

Reply:
Multistream uses a URI to provide info about a protocol, e. g. /http/w3id.org/ipfs/1.1.0 or /bittorrent.org/1.0 .